signer/src/main/java/com/uport/sdk/signer/UportHDSigner.kt

```diff
@@ -156,8 +156,8 @@ class UportHDSigner : UportSigner() {
                 val sigData = keyPair.signMessage(txBytes)
                 return@decrypt callback(null, sigData)
 
-            } catch (exception: Exception) {
-                return@decrypt callback(exception, EMPTY_SIGNATURE_DATA)
+            } catch (signError: Exception) {
```

Comment:
The error caught here is a generic `Exception`, is the aim to specifically catch a `SignError`? Is it guaranteed that no other errors can occur here?

Reply:
The idea is not to handle specific exceptions but to funnel them along into the callback method.
The different variable name is there just to differentiate between the multiple levels where the callback can be called from.